src/Network/Cookie/RequestCookieJar.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Cake\Network\Cookie;
+
+use Cake\Core\Configure;
+
+class RequestCookieJar extends AbstractCookieJar
+{
+
+    /**
+     *
+     * @param array $cookies
+     */
+    public function __construct(array $cookies)
+    {
+        foreach ($cookies as $name => $cookieData) {
+            $cookieData += (array)Configure::read("Cookies.$name");
+            $cookieData['name'] = $name;
+
+            $cookie = new $this->_cookieClassName($cookieData);
+            $this->_cookies[$cookie->name()] = $cookie;
```

Comment:
@markstory RequestCookieJar doesn't have add() method. I;m wondering whether it should have it and should we be able to add cookies to te request on runtime.